02/intersection.md

```diff
@@ -0,0 +1,348 @@
+    * What is the expected output for the given input?
+    * You can use the examples provided in the prompt, or other examples.
+2. Write unit tests for `intersection_node` for the nominal and edge cases you identified in the first step.
+
+*Note: Click the **Run Tests** button to save your tests for instructor feedback. No real tests are actually run again your unit tests.*
+
+##### !end-question
+##### !placeholder
+
+```py
+# example input 1:
+# expected output 1:
+
+# example input 2:
+# expected output 2:
+
+def test_nominal_case():
+    # ^rename with meaningful test name
+    # and complete test implementation below
+    pass
+    # arrange
+
+    # act
+
+    # assert
+
+def test_edge_case():
+    # ^rename with meaningful test name
+    # and complete test implementation below
+    pass
+    
+    # arrange
+    
+    # act
+    
+    # assert
+```
+##### !end-placeholder
+
+##### !tests
+
+```py
+import unittest
+
+class TestPython1(unittest.TestCase):
+  def test_always_pass(self):
+    self.assertEqual(1,1)
+```
+
+##### !end-tests
+##### !explanation 
+
+Example tests:
+
+```python
+def test_will_return_intersection_for_lists_of_same_length():
+    # Arrange
+    node_d = Node("D")
+
+    node_x = Node("X")
+
+    node_one = Node("1")
+    node_two = Node("2")
+    node_three = Node("3")
+    node_one.next = node_two
+    node_two.next = node_three
+
+    # List A: D -> 1 -> 2 -> 3
+    node_d.next = node_one
+
+    # List B: X -> 1 -> 2 -> 3
+    node_x.next = node_one
+
+    head_a = node_d
+    head_b = node_x
+
+    # Act
+    answer = intersection_node(head_a, head_b)
+
+    # Assert
+    assert answer == node_one
+
+def test_will_return_none_with_one_empty_list():
+    # Arrange
+    node_d = Node("D")
+    node_e = Node("E")
+    node_f = Node("F")
+
+    # List A: D -> E -> F
+    node_d.next = node_e
+    node_e.next = node_f
+
+    # List B: None, the list is empty
+
+    # Act
+    answer = intersection_node(node_d, None)
+
+    # Assert
+    assert answer is None
+```
+
+##### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 3 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: paragraph
+* id: 10510a7b-e915-42cc-8cea-26555f96c34d
+* title: Create Logical Steps
+* topics: pse
+##### !question
+
+Without writing code, describe how you would implement `intersection_node` in enough detail that someone else could write the code. 
+* It may be helpful to break up the problem/algorithm into smaller subproblems/algorithms. For example, 1. Handle invalid input, 2. Given valid input, perform the computation/solve the problem/etc.
+* Your logical steps could take the form of a numbered list, pseudo code, or anywhere in between. What's important at this stage is to think through and outline the implementation before writing code.
+
+##### !end-question
+
+##### !placeholder
+
+Write the logical steps here.
+
+##### !end-placeholder
+
+### !end-challenge
```

Comment:
Did you make a decision not to add sample logical steps in the explanation like we do in core PSEs? I don't remember if we actually discussed this.

Reply:
Yes, I decided not to include the sample logical steps because I saw them as hints and wanted to challenge the interns to come up with logical steps from scratch. That being said, I can add them here (and to the Pascal's triangle IPQ as well) and folx can choose to access them whenever needed 🙌🏾